bin/prettier.js

```diff
@@ -30,45 +33,61 @@ if (!filenames.length) {
   process.exit(1);
 }
 
-filenames.forEach(filename => {
-  fs.readFile(filename, "utf8", (err, input) => {
-    if (write) {
-      console.log(file);
-    }
-
-    if (err) {
-      console.error("Unable to read file: " + filename + "\n" + err);
-      // Don't exit the process if one file failed
-      process.exitCode = 2;
-      return;
-    }
+function format(input) {
+  return jscodefmt.format(input, {
+    printWidth: argv["print-width"],
+    tabWidth: argv["tab-width"],
+    bracketSpacing: argv["bracket-spacing"],
+    useFlowParser: argv["flow-parser"],
+    singleQuote: argv["single-quote"],
+    trailingComma: argv["trailing-comma"]
+  });
+}
 
-    let output;
+if (stdin) {
+  getStdin().then(input => {
     try {
-      output = jscodefmt.format(input, {
-        printWidth: argv["print-width"],
-        tabWidth: argv["tab-width"],
-        bracketSpacing: argv["bracket-spacing"],
-        useFlowParser: argv["flow-parser"],
-        singleQuote: argv["single-quote"],
-        trailingComma: argv["trailing-comma"]
-      });
+      console.log(format(input));
     } catch (e) {
       process.exitCode = 2;
       console.error(e);
       return;
     }
+  });
+} else {
+  filenames.forEach(filename => {
+    fs.readFile(filename, "utf8", (err, input) => {
+      if (write) {
+        console.log(filename);
```

Comment:
FYI there was a bug here, it was `console.log(file)` before (which is not defined.